e2e_playwright/main_menu_test.py

```diff
@@ -22,3 +22,66 @@ def test_main_menu_images(themed_app: Page, assert_snapshot: ImageCompareFunctio
 
     element = themed_app.get_by_test_id("stMainMenuPopover")
     assert_snapshot(element, name="main_menu")
+
+
+def test_renders_settings_dialog_properly(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    app.get_by_test_id("stMainMenu").click()
+
+    app.get_by_text("Settings").click()
+    assert_snapshot(app.get_by_role("dialog"), name="settings_dialog")
+
+
+# This test doesn't work depending on webkit (safari) browser
+# as a result, going to comment this test out
+# def test_renders_screencast_dialog_properly(
+#     app: Page, assert_snapshot: ImageCompareFunction
+# ):
+#     app.get_by_test_id("stMainMenu").click()
+
+#     app.get_by_text("Record a screencast").click()
+#     assert_snapshot(app.get_by_role("dialog"), name="record_screencast_dialog")
+
+
+# This test doesn't work depending on webkit (safari) browser
+# as a result, going to comment this test out
+# def test_renders_screencast_recorded_dialog_properly(
+#     app: Page, assert_snapshot: ImageCompareFunction
+# ):
+#     app.get_by_test_id("stMainMenu").click()
+
+#     app.get_by_text("Record a screencast").click()
+#     app.get_by_text("Start recording!").click()
+
+#     # Wait 5 seconds because there is a 3! 2! 1! on the screen until recording occurs and there may be buffer
+#     app.wait_for_timeout(5000)
+
+#     # Remove the browser support dialog message
+#     app.keyboard.press("Escape")
+#     app.get_by_test_id("stMainMenu").click()
+
+#     app.get_by_text("Stop recording").click()
+
+#     # don't use screenshot as the recording may differ so just check for specific text
+#     expect(
+#         app.get_by_role("dialog").get_by_text("Preview your video below:")
+#     ).to_be_visible
+
+
+def test_renders_about_dialog_properly(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    app.get_by_test_id("stMainMenu").click()
+
+    app.get_by_text("About").click()
+    assert_snapshot(app.get_by_role("dialog"), name="about_dialog")
+
+
+def test_renders_clear_cache_dialog_properly(
+    app: Page, assert_snapshot: ImageCompareFunction
+):
+    app.get_by_test_id("stMainMenu").click()
+
+    app.get_by_text("Clear cache").click()
+    assert_snapshot(app.get_by_role("dialog"), name="clear_cache_dialog")
```

Comment:
I have seen some flakiness previously related to modals being opened and screenshots being taken (with the deploy dialogs). I think we should add more code here to make sure that the dialog is displayed correctly before we take screenshots:

```suggestion
    dialog = app.get_by_test_id("stModal")
    expect(dialog).to_be_visible()
    expect(dialog).to_have_text(r".*Clear caches.*")
    assert_snapshot(dialog, name="clear_cache_dialog")
```

The same is also with the other dialogs.